itests/eth_transactions_test.go

```diff
@@ -0,0 +1,299 @@
+		V:                    big.Zero(),
+		R:                    big.Zero(),
+		S:                    big.Zero(),
+	}
+
+	unsigned, err := tx.ToRlpUnsignedMsg()
+	require.NoError(t, err)
+	// Submit transaction without signing message
+	_, err = client.EVM().EthSendRawTransaction(ctx, unsigned)
+	require.Error(t, err)
+
+	client.EVM().SignTransaction(&tx, key.PrivateKey)
+
+	hash := client.EVM().SubmitTransaction(ctx, &tx)
+
+	receipt, err := waitForEthTxReceipt(ctx, client, hash)
+	require.NoError(t, err)
+	require.NotNil(t, receipt)
+
+	// Success.
+	require.EqualValues(t, ethtypes.EthUint64(0x1), receipt.Status)
+
+	// Verify that the deployer is now an account.
+	client.AssertActorType(ctx, deployer, manifest.EthAccountKey)
+
+	// Verify that the nonce was incremented.
+	nonce, err := client.MpoolGetNonce(ctx, deployer)
+	require.NoError(t, err)
+	require.EqualValues(t, 1, nonce)
+
+	// Verify that the deployer is now an account.
+	client.AssertActorType(ctx, deployer, manifest.EthAccountKey)
+}
+
+func TestContractInvocation(t *testing.T) {
+	blockTime := 100 * time.Millisecond
+	client, _, ens := kit.EnsembleMinimal(t, kit.MockProofs(), kit.ThroughRPC())
+
+	ens.InterconnectAll().BeginMining(blockTime)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	// install contract
+	contractHex, err := os.ReadFile("./contracts/SimpleCoin.bin")
+	require.NoError(t, err)
+
+	contract, err := hex.DecodeString(string(contractHex))
+	require.NoError(t, err)
+
+	// create a new Ethereum account
+	key, ethAddr, deployer := client.EVM().NewAccount()
+	// send some funds to the f410 address
+	kit.SendFunds(ctx, t, client, deployer, types.FromFil(10))
+
+	// DEPLOY CONTRACT
+
+	gaslimit, err := client.EthEstimateGas(ctx, ethtypes.EthCall{
+		From: &ethAddr,
+		Data: contract,
+	})
+	require.NoError(t, err)
+
+	maxPriorityFeePerGas, err := client.EthMaxPriorityFeePerGas(ctx)
+	require.NoError(t, err)
+
+	// now deploy a contract from the embryo, and validate it went well
+	tx := ethtypes.EthTxArgs{
+		ChainID:              build.Eip155ChainId,
+		Value:                big.Zero(),
+		Nonce:                0,
+		MaxFeePerGas:         types.NanoFil,
+		MaxPriorityFeePerGas: big.Int(maxPriorityFeePerGas),
+		GasLimit:             int(gaslimit),
+		Input:                contract,
+		V:                    big.Zero(),
+		R:                    big.Zero(),
+		S:                    big.Zero(),
+	}
+
+	client.EVM().SignTransaction(&tx, key.PrivateKey)
+
+	hash := client.EVM().SubmitTransaction(ctx, &tx)
+	fmt.Println(hash)
+
+	receipt, err := waitForEthTxReceipt(ctx, client, hash)
+	require.NoError(t, err)
+	require.NotNil(t, receipt)
+	require.EqualValues(t, ethtypes.EthUint64(0x1), receipt.Status)
+
+	// Get contract address.
+	contractAddr := client.EVM().ComputeContractAddress(ethAddr, 0)
+
+	// INVOKE CONTRACT
+
+	params, err := hex.DecodeString("f8b2cb4f000000000000000000000000ff00000000000000000000000000000000000064")
```

Comment:
Comment describing what's happening here (which method's being invoked, what the params are, and maybe a link to the Solidity ABI)